internal/lint/html_test.go

```diff
@@ -0,0 +1,167 @@
+package lint
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/errata-ai/vale/v3/internal/core"
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_applyPatterns(t *testing.T) {
+	cases := []struct {
+		description string
+		conf        core.Config
+		exts        extensionConfig
+		content     string
+		expected    string
+	}{
+		{
+			description: "MDX comment in markdown, custom comment delimiter",
+			conf: core.Config{
+				CommentDelimiters: map[string][2]string{
+					".md": [2]string{"{/*", "*/}"},
+				},
+			},
+			exts: extensionConfig{".md", ".md"},
+			content: `---
+title: Example page
+description: Example page
+---
+
+This is the intro pagragraph.
+
+{/* This is a comment */}
+`,
+			expected: strings.ReplaceAll(`
+@@@
+title: Example page
+description: Example page
+@@@
+
+
+This is the intro pagragraph.
+
+<!-- This is a comment -->
+`, "@", "`"),
+		},
+		{
+			description: "MDX comment in markdown, no custom comment delimiter",
+			conf:        core.Config{},
+			exts:        extensionConfig{".md", ".md"},
+			content: `---
+title: Example page
+description: Example page
+---
+
+This is the intro pagragraph.
+
+{/* This is a comment */}
+`,
+			expected: strings.ReplaceAll(`
+@@@
+title: Example page
+description: Example page
+@@@
+
+
+This is the intro pagragraph.
+
+{/* This is a comment */}
+`, "@", "`"),
+		},
+		{
+			description: "multiline MDX comment in markdown, custom comment delimiter",
+			conf: core.Config{
+				CommentDelimiters: map[string][2]string{
+					".md": [2]string{"{/*", "*/}"},
+				},
+			},
+			exts: extensionConfig{".md", ".md"},
+			content: `---
+title: Example page
+description: Example page
+---
+
+This is the intro pagragraph.
+
+{/* 
+This is a comment 
+*/}
+`,
+			expected: strings.ReplaceAll(`
+@@@
+title: Example page
+description: Example page
+@@@
+
+
+This is the intro pagragraph.
+
+<!-- 
+This is a comment 
+-->
+`, "@", "`"),
+		},
+		{
+			description: "token ignore in cc file",
+			content:     "Call \\c func to start the process.",
+			conf: core.Config{
+				TokenIgnores: map[string][]string{
+					"*.cc": []string{"(\\\\c \\w+)"},
```

Comment:
When I see such a thing, I always wonder if using string literals (backtick notation) won't help

Something like

```suggestion
					"*.cc": []string{`(\\c \w+)`},
```

Or this


```suggestion
					"*.cc": []string{"(\c \w+)"},
```

Because I'm unsure why you could need a double escaped c, but a simple escaped w

Reply:
Thanks! Fixed in [7ed02e6f92b311dff574f2eb57addc78f0b47eec](https://github.com/errata-ai/vale/pull/900/commits/7ed02e6f92b311dff574f2eb57addc78f0b47eec) . I think I'd copied a value from `dlv` to write the test and didn't consider cleaning it up.